test/manipulation.test.js

```diff
@@ -901,6 +902,103 @@ describe('manipulation', function () {
         e.should.be.eql(new Error('Invalid date: X'));
       }
     });
+  });
+
+  describe('update|updateAll', function() {
+    beforeEach(function destroyFixtures(done) {
+      Person.destroyAll(done);
+    });
+
+    beforeEach(function createFixtures(done) {
+      var people = [{
+        name: 'foo',
+        age: 19
+      }, {
+        name: 'bar',
+        age: 20
+      }, {
+        name: 'baz',
+        age: 21
+      }, {
+        name: 'qux',
+        age: 22
+      }, {
+        name: 'quux',
+        age: 23
+      }];
+      async.each(people, Person.create.bind(Person), done);
+    });
+
+    it('should be a function', function() {
+      Person.update.should.be.a.Function;
+      Person.updateAll.should.be.a.Function;
+    });
+
+    it('should not update instances that do not satisfy the where condition',
+        function(done) {
+      Person.update({name: 'corge'}, {name: 'grault'}, function(err, count) {
+        should.not.exist(err);
+        count.should.equal(0);
+        Person.find({where: {name: 'corge'}}, function(err, people) {
+          should.not.exist(err);
+          people.should.be.empty;
+          done();
+        });
+      });
+    });
+
+    it('should update instances that satisfy the where condition',
+        function(done) {
+      Person.update({name: 'foo'}, {name: 'bar'}, function(err, count) {
+        should.not.exist(err);
+        count.should.equal(1);
+        Person.find({where: {age: 19}}, function(err, people) {
+          should.not.exist(err);
+          people.should.have.length(1);
+          people[0].name.should.equal('bar');
+          done();
+        });
+      });
+    });
+
+    it('should update all instances when the where condition is not provided',
+        function(done) {
+      Person.update({name: 'corge'}, function(err, count) {
+        should.not.exist(err);
+        count.should.equal(5);
+        Person.find({where: {name: 'foo'}}, function(err, people) {
+          should.not.exist(err);
+          people.should.be.empty;
+          Person.find({where: {name: 'corge'}}, function(err, people) {
+            should.not.exist(err);
+            people.should.have.length(5);
+            done();
+          });
+        });
+      });
+    });
 
+    it('should ignore where conditions with undefined values',
+        function(done) {
+      Person.update({name: 'foo'}, {name: undefined}, function(err, count) {
```

Comment:
This test will fail with mongodb. The data needs to have at least one property value. `{name: undefined}` will becomes `{}`.